cdap-common/src/main/resources/cdap-default.xml

```diff
@@ -2317,4 +2317,342 @@
+  </property>
+
+  <property>
+    <name>spark.retry.policy.max.time.secs</name>
+    <value>600</value>
+    <description>
+      The max elapsed time in seconds before retries are aborted.
+    </description>
+  </property>
+
+  <property>
+    <name>spark.retry.policy.max.retries</name>
+    <value>1000</value>
+    <description>
+      The max number of retries to attempt before aborting.
+      By default, the max time will be hit before this will be hit
+    </description>
+  </property>
+
+  <property>
+    <name>spark.retry.policy.delay.base.ms</name>
+    <value>1000</value>
+    <description>
+      The base delay between retries in milliseconds. Used by the 'fixed.delay' and 'exponential.backoff' policy types
+    </description>
+  </property>
+
+  <property>
+    <name>spark.retry.policy.delay.max.ms</name>
+    <value>30000</value>
+    <description>
+      The maximum delay between retries in milliseconds. Only used if the policy type is 'exponential.backoff'
+    </description>
+  </property>
+
+  <!-- Custom Action configuration -->
+
+  <property>
+    <name>custom.retry.policy.type</name>
+    <value>exponential.backoff</value>
+    <description>
+      The type of retry policy. Must be 'none', 'fixed.delay', or 'exponential.backoff'.
+      The retry policy is used when CDAP attempted to perform an operation, but that operation failed in a retryable
+      way. The most common type of failure is if a dependent service was unavailable. For example, if a transaction
+      could not be started because the transaction service is not available. Any CDAP method that throws a
+      RetryableException will use this policy if it encounters a RetryableException
+    </description>
+  </property>
+
+  <property>
+    <name>custom.retry.policy.max.time.secs</name>
+    <value>600</value>
+    <description>
+      The max elapsed time in seconds before retries are aborted.
+    </description>
+  </property>
+
+  <property>
+    <name>custom.retry.policy.max.retries</name>
+    <value>1000</value>
+    <description>
+      The max number of retries to attempt before aborting.
+      By default, the max time will be hit before this will be hit
+    </description>
+  </property>
+
+  <property>
+    <name>custom.retry.policy.delay.base.ms</name>
+    <value>1000</value>
+    <description>
+      The base delay between retries in milliseconds. Used by the 'fixed.delay' and 'exponential.backoff' policy types
+    </description>
+  </property>
+
+  <property>
+    <name>custom.retry.policy.delay.max.ms</name>
+    <value>30000</value>
+    <description>
+      The maximum delay between retries in milliseconds. Only used if the policy type is 'exponential.backoff'
+    </description>
+  </property>
+
+  <!-- Workflow configuration -->
+
+  <property>
+    <name>workflow.retry.policy.type</name>
+    <value>exponential.backoff</value>
+    <description>
+      The type of retry policy. Must be 'none', 'fixed.delay', or 'exponential.backoff'.
```

Comment:
I agree that this will be difficult to maintain. As I read this, I can also see other properties (all of the `custom.action.retry.*` properties) that also reference this, and should reference this. They too refer to these policies by name, and thus become fragile. 

Ideas:

- One solution is use numbers instead of names. Then, the `type` is an integer (0, 1, 2, etc. or some other number, if other systems use a similar strategy) whose meaning is defined elsewhere, in the main CDAP documentation. Add another integer with another meaning becomes trivial. See http://docs.aws.amazon.com/general/latest/gr/api-retries.html for an example using an enum.

- My sense is that there is **too much documentation here** and **none of this commentary should be here**. It should all be moved to the documentation, with a reference as appropriate. The reference could be in the section divider (`<!-- "title" configuration -->`), though I'd need to change the code that generates the documentation table to handle comments. Not a big issue, but just something that would need to be done.